server/service/src/stocktake_line/insert.rs

```diff
@@ -201,6 +204,24 @@ fn validate(
             return Err(InsertStocktakeLineError::AdjustmentReasonNotValid);
         }
     }
+
+    if let (Some(counted_number_of_packs), Some(stock_line)) =
+        (input.counted_number_of_packs, &stock_line)
+    {
+        if check_stock_line_reduced_below_zero(
+            connection,
+            store_id,
+            &stock_line.stock_line_row.id,
+            &counted_number_of_packs,
+        )? {
+            return Err(InsertStocktakeLineError::StockLineReducedBelowZero(
```

Comment:
As per the other comment, this should provide stock line, then it can be used in API and returned to front end